jedi/api/__init__.py

```diff
@@ -108,7 +110,11 @@ def __init__(self, source=None, line=None, column=None, path=None,
         self._user_context = UserContext(self.source, self._pos)
         self._parser = UserContextParser(self._grammar, self.source, path,
                                          self._pos, self._user_context)
-        self._evaluator = Evaluator(self._grammar)
+        if sys_path is None:
+            venv = os.getenv('VIRTUAL_ENV')
+            if venv:
+                sys_path = list(get_venv_path(venv)) + sys.path
```

Comment:
> most of the time it will refer to one or more virtualenvs, so probably specifying virtualenv paths will look better. 

Why would it refer to multiple virtualenvs? That's not something that happens in Python itself?

> If this API addition goes through, I'm not sure if it is worth keeping the VIRTUAL_ENV logic: it boils down to having both VIRTUAL_ENV's and jedi's sys.path in one interpreter and this is better accomplished the other way around. I mean, that logic starts with jedi on sys.path and then tries to add venv path afterwards,

Hmm I know what you mean. I personally think that adding `sys_path` as an API is really valuable, because some people really want to play with it. Most of the time this is not the case and we should focus on what happens when it is `None`. This still includes virtual envs. However, I agree totally with the `portable.py` idea. I think Jedi should be started with just the virtualenv as its path. This is what I'm talking about when I talk about Jedi's future async server/client structure. You name a certain virtualenv and Jedi starts a new process that includes that virtualenv. This way, we don't have segfaults and all other crazy stuff...

What do you think? This would however mean that we don't include this pull request, we would rather just add the sys_path logic. And BTW this is not to say that your work is bad. I'm really thankful for it. It taught me a lot! :)

~ Dave